easybuild/easyblocks/g/gromacs.py

```diff
@@ -324,34 +378,61 @@ def configure_step(self):
                     if not regex.search(out):
                         raise EasyBuildError("Pattern '%s' not found in GROMACS configuration output.", pattern)
 
+    def build_step(self):
+        """
+        Custom build step for GROMACS; Skip if CUDA is enabled and the current
+        iteration is for double precision
+        """
+
+        if self.is_double_precision_cuda_build():
+            self.log.info("skipping build step")
+            return
+
+        super(EB_GROMACS, self).build_step()
+
     def test_step(self):
         """Run the basic tests (but not necessarily the full regression tests) using make check"""
+
+        if self.is_double_precision_cuda_build():
+            self.log.info("skipping test step")
+            return
+
         # allow to escape testing by setting runtest to False
-        if not self.cfg['runtest'] and not isinstance(self.cfg['runtest'], bool):
+        if self.cfg['runtest'] is None or self.cfg['runtest']:
 
+            old_runtest = self.cfg['runtest']
             # make very sure OMP_NUM_THREADS is set to 1, to avoid hanging GROMACS regression test
             env.setvar('OMP_NUM_THREADS', '1')
 
-            self.cfg['runtest'] = 'check'
-            if self.cfg['parallel']:
-                # run 'make check' in parallel since it involves more compilation
-                self.cfg.update('runtest', "-j %s" % self.cfg['parallel'])
+            if isinstance(self.cfg['runtest'], bool):
+                self.cfg['runtest'] = 'check'
```

Comment:
I think this should also be done if `runtest` is `Non`e, otherwise the `update` below will fail in that case

Reply:
Actually: Why the condition at all? Why not just `self.cfg['runtest'] = 'check -j %s' % self.cfg['parallel']`?
I get that it may allow runtest to be a string with a custom target but that hasn't been the case before and I don't see why it is needed

Reply:
I made it to make it easy to pick another make target if they decide to use one.